src/spatial/nearest_neighbor.cpp

```diff
@@ -0,0 +1,95 @@
+
+#include "cliques/cliquer/graph.h"
+#include "igraph_operators.h"
+#include "igraph_spatial.h"
+#include "igraph_datatype.h"
+#include "igraph_matrix.h"
+#include "igraph_error.h"
+#include "igraph_types.h"
+#include "nanoflann/nanoflann.hpp"
+#include <cstddef>
+#include <cstdint>
+#include <string.h>
+
+class igraph_point_adaptor {
+    igraph_integer_t dimension;
+    const igraph_matrix_t *points;
+    size_t kdtree_get_point_count() const {
+        return igraph_matrix_nrow(points);
+    }
+    igraph_real_t kdtree_get_pt(const size_t idx, const size_t dim) const {
+        return igraph_matrix_get(points, idx, dim);
```

Comment:
Do not use `matrix_get()` and `vector_get()`, as the performance is bad. Use `MATRIX` and `VECTOR` instead, which are macros that can be inlined.